RecoPixelVertexing/PixelLowPtUtilities/bin/ClusterShapeAnalyzer.cc

```diff
@@ -0,0 +1,400 @@
+  {
+    for(int d=0; d<2; d++) // direction
+    if(x[b][d]->Integral() > 0)
+    {
+      double limits[2] = {0, 0};
+      double dmin = 1e+9;
+
+      for(double f = (Loss/2)/100; f < Loss/2 - (Loss/2)/200; f+=(Loss/2)/100)
+      {
+        double x0 = getFraction(x[b][d],               f );
+        double x1 = getFraction(x[b][d], 1 - (Loss/2 - f));
+
+        if(fabs(x1 - x0) < dmin)
+        {
+          limits[0] = x0;
+          limits[1] = x1;
+
+          dmin = fabs(x1 - x0);
+        }
+      }
+
+      c[b].lim[d][0] = limits[0];
+      c[b].lim[d][1] = limits[1];
+//      c[b].lim[d][0] = getFraction(x[b][d],     Loss/2);
+//      c[b].lim[d][1] = getFraction(x[b][d], 1 - Loss/2);
+    }
+
+    // Area
+    c[b].area = (c[b].lim[0][1] - c[b].lim[0][0]) *
+                (c[b].lim[1][1] - c[b].lim[1][0]);
+  }
+}
+
+/****************************************************************************/
+void ClusterAnalyzer::printOut
+  (TH2F * histo, vector<Point>& points, vector<Center>& c, char* flag)
+{
+  char fileName[256];
+  ofstream outFile;
+
+  //////////////////////////////////
+  sprintf(fileName,"../out/points_%s",flag);
+  printToFile(histo, fileName);
+
+  //////////////////////////////////
+  sprintf(fileName,"../out/limits_%s.par",flag);
+  outFile.open(fileName);
+
+  char label[128];
+  if(histo->Integral() >= MinEntries)
+    sprintf(label," ");
+  else
+    sprintf(label,"Too few entries, not used");
+  
+
+  for(int b=0; b<2; b++)
+  for(int d=0; d<2; d++)
+  {
+    outFile << " l" << b << d << "=" << c[b].lim[d][0] << endl;
+    outFile << " h" << b << d << "=" << c[b].lim[d][1] << endl;
+  }
+
+  outFile.close(); 
+
+  //////////////////////////////////
+  sprintf(fileName,"../out/centers_%s.dat",flag);
+  outFile.open(fileName);
+
+  for(int b=0; b<2; b++)
+    outFile << " " << c[b].x[0] << " " << c[b].x[1] << endl;
+
+  outFile.close();
+
+  //////////////////////////////////
+//  *g << "call '../gnu/pixelShape.gnu' '" << flag << "' '" << label << "'\n";
+}
+
+/****************************************************************************/
+void ClusterAnalyzer::process(TH2F * histo, vector<Center>& c,
+                              char* flag)
+{
+  vector<Point> points;
+
+  for(int i = 1 ; i <= histo->GetNbinsX(); i++)
+  for(int j = 1 ; j <= histo->GetNbinsY(); j++)
+    if(histo->GetBinContent(i,j) > 0)
+    {
+      Point p;
+      p.x[0] = histo->GetXaxis()->GetBinCenter(i);
+      p.x[1] = histo->GetYaxis()->GetBinCenter(j);
+      p.w    = int(histo->GetBinContent(i,j) + 0.5);
+
+      points.push_back(p);
+    }
+
+  line = histo->ProjectionY("_py",0,0);
+  line->Reset();
+
+  kMeans    (points,c);
+  findLimits(points,c);
+  printOut  (histo, points,c, flag);
+}
+
+/****************************************************************************/
+void ClusterAnalyzer::analyze(const std::string& inputFileName, const std::string& outputFileName)
+{
+    static const char *subName[2] = {"barrel","endcap"};
```

Comment:
it can be just made in local scope (drop ```static```)
or changed to ```static const char* const subName``` to not disturb the static analyzer